src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.State.cs

```diff
@@ -302,6 +211,42 @@ private static string ConcatNamespace(string rootNamespace, string namespaceSuff
                     return rootNamespace + "." + namespaceSuffix;
                 }
             }
+
+            /// <summary>
+            /// Try get the relative namespace for <paramref name="namespace"/> based on <paramref name="relativeTo"/>,
+            /// if <paramref name="relativeTo"/> is the containing namespace of <paramref name="namespace"/>. Otherwise,
+            /// Returns null.
+            /// For example:
+            /// - If <paramref name="relativeTo"/> is "A.B" and <paramref name="namespace"/> is "A.B.C.D", then
+            /// the relative namespace is "C.D".
+            /// - If <paramref name="relativeTo"/> is "A.B" and <paramref name="namespace"/> is also "A.B", then
+            /// the relative namespace is "".
+            /// - If <paramref name="relativeTo"/> is "" then the relative namespace us <paramref name="namespace"/>.
+            /// </summary>
+            private static string GetRelativeNamespace(string relativeTo, string @namespace, ISyntaxFactsService syntaxFacts)
```

Comment:
It doesn't really morally feel like a snytax fact... 

It's actually unclear why this would need ISyntaxFactsService in teh first place.  If this is just about case-sensitivity, then that feels like it should just be a boolean that is passed in IMO.

Reply:
I feel this is doing something useful only in a very limited scope (i.e. this service), don't want to promote it until there's a need for it.

Reply:
I see. What's the benefit of using the `ISyntaxFactsService.StringComparer` for cases of equality? It looks like it just uses `StringComparer.Ordinal` . Does the comparison need to align with `ISyntaxFactsService`?

Reply:
In general, when you can use `ISyntaxFactsService`, you do that to avoid duplicate code. Additionally this this case, when the refactoring was implemented, we still had VB in mind. Even though we later decided not to provide this for VB, the class hierarchy remained the same, just in case we change our mind in the future.And `ISyntaxFactsService.StringComparer` can dealt with VB as well.